src/JobPanel/backend/json_data.py

```diff
@@ -0,0 +1,176 @@
+import json
+
+
+class ClassFactory:
+    """
+    Helper class for JsonData.
+    """
+    def __init__(self, class_list):
+        """
+        Initialize list of possible classes.
+        :param class_list:
+        """
+        self.class_list = class_list
+
+    def make_instance(self, config):
+        """
+        Make instance from config dict.
+        Dict must contain item "__class__" with name of desired class.
+        Desired class must be in class_list.
+        :param config:
+        :return:
+        """
+        if "__class__" not in config:
+            return None
+
+        for c in self.class_list:
+            if c.__name__ == config["__class__"]:
+                return c(config)
+        return None
+
+class JsonData:
+    """
+    Abstract base class for various data classes.
+    These classes are basically just documented dictionaries,
+    which are JSON serializable and provide some syntactic sugar
+    (see DotDict from Flow123d - Jan Hybs)
+    In order to simplify also serialization of non-data classes, we
+    should implement serialization of __dict__.
+
+    Why use JSON for serialization? (e.g. instead of pickle)
+    We want to use it for both sending the data and storing them in files,
+    while some of these files should be human readable/writable.
+
+    Serializable classes will be derived from this one. And data members
+    that should not be serialized are prefixed by '_'.
+
+    ?? Anything similar in current JobPanel?
+    """
+    def __init__(self, config):
+        """
+        Initialize class dict from config serialization.
+        :param config:
+        """
+        for k, v in config.items():
+            if k in self.__dict__:
+                # dict
```

Comment:
Can we make just one check for type match here? e.g. 

	self_v = self.__dict__[k]
	assert( self_v.__class__ == v.__class )

ClassFactory must be checked before this check